src/modules/familyMember/application/familyMember.service.ts

```diff
@@ -62,29 +61,67 @@ export class FamilyMemberService {
         if (!existingFamilyMember) {
             throw new NotFoundException(`Family member with ID ${id} not found.`);
         }
+
         if (dto.registrationNumber) {
             const regOwner = await this.familyMemberRepository.findByRegistrationNumber(dto.registrationNumber);
             if (regOwner && regOwner.getId() !== id){
-                throw new ConflictException("CPF '${dto.registrationNumber}' already in use!");
+                throw new ConflictException(`CPF '${dto.registrationNumber}' is already in use.`);
             }
         }
         if (dto.email) {
             const emailOwner = await this.familyMemberRepository.findByEmail(dto.email);
             if (emailOwner && emailOwner.getId() !== id) {
-                throw new ConflictException(`Email '${dto.email}' already in use!`);
+                throw new ConflictException(`Email '${dto.email}' is already in use.`);
             }
         }
         
-        const partialEntity = new FamilyMemberEntity({ id, ...existingFamilyMember, ...dto } as any);
+        Object.keys(dto).forEach(key => {
+            const setterName = `set${key.charAt(0).toUpperCase() + key.slice(1)}`;
+            if (typeof existingFamilyMember[setterName] === 'function') {
+                existingFamilyMember[setterName](dto[key]);
+            }
+        });
+        
+        return this.familyMemberRepository.update(existingFamilyMember);
+    }
 
-        return this.familyMemberRepository.update(partialEntity);
+    async addAddressToFamilyMember(familyMemberId: number, dto: CreateAddressDTO): Promise<AddressEntity> {
+        const familyMember = await this.findById(familyMemberId);
+        
```

Comment:
```suggestion
        if(familyMember.getAddressId() != null) {
            throw new ConflictException(
                `Family member with ID ${familyMember.getId()} already has an address in the database.`
            );
}
```
Sugiro só deixar inserir um novo endereço, se o familyMember não tiver um endereço previamente cadastrado. Do contrário, a tabela `address` vai ficar com endereços sem um familyMember associado.

Reply:
certo, ajustei